chajda/tsquery/augments.py

```diff
@@ -46,10 +67,51 @@ def augments_gensim(lang, word, config=Config(), n=5):
     return words
 
 
-# load the model if it's not already loaded
-try:
-    augments_gensim.model
-except AttributeError:
-    import gensim.downloader
-    augments_gensim.model = gensim.downloader.load("glove-wiki-gigaword-50")
+import fasttext
+import fasttext.util
+
+fasttext_models = {}
+
+def augments_fasttext(lang, word, config=Config(), n=5):
+    ''' 
+    Returns n words that are "similar" to the input word in the target language.
+    These words can be used to augment a search with the Query class.
+
+    Note: Due to the size of fasttext models, testing multiple languages in the doctest requires more space than github actions allows for. For this reason, only english is tested in the doctest.
+
+    >>> to_tsquery('en', 'baby boy', augment_with=augments_fasttext)
+    '(baby:A | newborn:B | infant:B) & (boy:A | girl:B | boyhe:B | boyit:B)'
+
+    >>> to_tsquery('en', '"baby boy"', augment_with=augments_fasttext)
+    'baby:A <1> boy:A'
+
+    >>> to_tsquery('en', '"baby boy" (school | home) !weapon', augment_with=augments_fasttext)
+    '(baby:A <1> boy:A) & ((school:A | schoo:B | schoolthe:B | schoool:B | kindergarten:B) | (home:A | house:B | homethe:B | homewhen:B | homethis:B)) & !(weapon:A | weaponthe:B | weopon:B)'
 
+    >>> augments_fasttext('en','weapon', n=5)
+    ['weaponthe', 'weopon']
+
+    >>> augments_fasttext('en','king', n=5)
+    ['queen', 'kingthe']
+    '''
+
+    try:
+        fasttext_models[lang]
+    except:
+        with suppress_stdout_stderr():
+            fasttext.util.download_model(lang, if_exists='ignore')
+        fasttext_models[lang] = fasttext.load_model('cc.{0}.300.bin'.format(lang))
+
+    #find the most similar words
+    try:
+        topn = fasttext_models[lang].get_nearest_neighbors(word, k=n)
+        words = ' '.join([ word for (rank, word) in topn ])
+    except KeyError:
```

Comment:
What is the cause of the `KeyError`?

My guess is that this is caused by `fasttext_models[lang]` not containing the `lang` key.  This should only happen if the model couldn't be downloaded.  In that case, we should probably raise a `ValueError` to indicate that the specified language is incorrect.